cstar/roms/runtime_settings.py

```diff
@@ -0,0 +1,806 @@
+from os import PathLike
+from pathlib import Path
+from typing import ClassVar, Optional, Union, get_args, get_origin
+
+from pydantic import (
+    BaseModel,
+    Field,
+    ModelWrapValidatorHandler,
+    model_serializer,
+    model_validator,
+)
+
+from cstar.base.log import get_logger
+from cstar.base.utils import _list_to_concise_str
+
+log = get_logger(__name__)
+
+################################################################################
+# Formatting methods for serializer:
+
+
+def _format_list_of_floats(float_list: list[float]) -> str:
+    joiner = " "
+    return joiner.join(_format_float(x) for x in float_list)
+
+
+def _format_list_of_paths(
+    path_list: list[Path], multi_line: Optional[bool] = False
+) -> str:
+    joiner = "\n    " if multi_line else " "
+    return joiner.join(_format_path(x) for x in path_list)
+
+
+def _format_list_of_other(other_list: list[str | int]) -> str:
+    joiner = " "
+    return joiner.join(_format_other(x) for x in other_list)
+
+
+def _format_float(val) -> str:
+    if val == 0.0:
+        return "0."
+    elif abs(val) < 1e-2 or abs(val) >= 1e4:
+        return f"{val:.6E}".replace("E+00", "E0")
+    else:
+        return str(val)
+
+
+def _format_path(path: Path) -> str:
+    return str(path)
+
+
+def _format_other(other: str | int) -> str:
+    return str(other)
+
+
+################################################################################
+
+
+class ROMSRuntimeSettingsSection(BaseModel):
+    section_name: ClassVar[str]
+    multi_line: ClassVar[bool] = False
+    key_order: ClassVar[list[str]]
+
+    def __init__(self, *args, **kwargs):
+        if args:
+            super().__init__(**{k: args[i] for i, k in enumerate(self.key_order)})
```

Comment:
a hard-coded key order is too easy to fall out of sync with model changes. consider making these init methods keyword only so we can get rid of ordering